File: 1913 B. Swap and Delete/Solution.cpp
// 321394683
#include<bits/stdc++.h> 
using namespace std;
// Krishno_Mondol_KUET_CSE
#define ll long long
void solve(){
    string str; cin>>str;
    int count_1=0, count_0=0;
    for(char ch : str){
        if(ch=='1'){count_1++;}
        else{count_0++;}
    }
    if(count_0==count_1){cout<<"0\n"; return;}
    for(char c : str){
        //if(count_0==0 || count_1==0){break;}
        if(c=='1'){
            if(count_0>0){count_0--;}
            else{break;}
        }
        else{
            if(count_1>0){count_1--;}
            else{break;}
        }
    }
    cout<<count_0+count_1<<endl;
}
int main(){int t; cin>>t; while(t--){solve();} return 0;}

File: 1913 B. Swap and Delete/Solution.py
# 321409086
# Krishno_Mondol_KUET_CSE
def solve(): 
    str = input()
    cnt_1,cnt_0 = 0,0
    for i in str:
        if i=='1': cnt_1+=1
        else: cnt_0+=1
    if cnt_0==cnt_1:
        print("0")
        return
    for i in str:
        if i=='1':
            if cnt_0>0: cnt_0-=1
            else: break
        else:
            if cnt_1>0: cnt_1-=1
            else: break
    print(cnt_1+cnt_0)

t = int(input())
for i in range(t):
    solve()
